### DISCUSSION TITLE
How to disable a refetch on mount when the query has already errored

### DISCUSSION THREAD

**Discussion by sophia-2b** [TanStack/query#5251] [Q&A]
I'm trying to work around the following behavior which seems intentional:

1. Multiple components all using the same query have been rendered, attempted to fetch, and are in an errored state.
2. An additional component gets mounted which uses that same query.
3. That newly mounted component attempts to fetch, resulting in the other components switching back to a loading state.

In step 3, I **don't** want that newly mounted component to attempt a fetch, but rather render the error state.

I'm assuming since the query has never succeeded, this technically isn't a "refetch", so the following options have no effect:

```
staleTime: Infinity,
refetchOnReconnect: "always",
refetchOnMount: false,
refetchOnWindowFocus: false,
retry: false,
```

The only hacky option I can think of is setting `enabled` to `false`, checking if the query is in an errored state, then forcing a refetch.. which then disables the auto-fetch I still want for the other components using this query.

Are there any other options I missed in the documentation for this desired behavior, or cleaner workarounds to achieve this?

**Comment by TkDodo** [comment]
you want `retryOnMount: false`

https://tanstack.com/query/v4/docs/react/reference/useQuery

> retryOnMount: boolean
If set to false, the query will not be retried on mount if it contains an error. Defaults to true.
